File: sdk/python/aura/types/compliance.py
"""Type definitions for Compliance module."""

from dataclasses import dataclass
from typing import Optional, List, Dict, Any
from enum import Enum
from datetime import datetime


class ComplianceStatus(Enum):
    """Compliance check status."""

    APPROVED = "approved"
    PENDING = "pending"
    REJECTED = "rejected"
    UNDER_REVIEW = "under_review"


class ComplianceLevel(Enum):
    """Compliance level."""

    BASIC = "basic"
    STANDARD = "standard"
    ENHANCED = "enhanced"


@dataclass
class ComplianceCheckParams:
    """Parameters for compliance check."""

    address: str
    check_type: str
    level: ComplianceLevel
    metadata: Optional[Dict[str, Any]] = None


@dataclass
class ComplianceReport:
    """Compliance report."""

    report_id: str
    address: str
    status: ComplianceStatus
    level: ComplianceLevel
    checks_passed: List[str]
    checks_failed: List[str]
    risk_score: float
    created_at: datetime
    expires_at: Optional[datetime] = None
    notes: Optional[str] = None


@dataclass
class KYCAMLStatus:
    """KYC/AML status."""

    address: str
    kyc_verified: bool
    aml_cleared: bool
    verification_date: Optional[datetime] = None
    expiry_date: Optional[datetime] = None
    provider: Optional[str] = None


@dataclass
class SanctionCheck:
    """Sanction screening result."""

    address: str
    is_sanctioned: bool
    lists_matched: List[str]
    checked_at: datetime
    details: Optional[str] = None


@dataclass
class TransactionMonitoring:
    """Transaction monitoring data."""

    tx_hash: str
    address: str
    amount: str
    risk_level: str
    flags: List[str]
    timestamp: datetime
    reviewed: bool = False

File: sdk/javascript/src/types/compliance.ts
/**
 * Compliance status enum
 */
export enum ComplianceStatusType {
  UNKNOWN = 0,
  PENDING = 1,
  APPROVED = 2,
  REJECTED = 3,
  REVOKED = 4,
}

/**
 * KYC level enum
 */
export enum KYCLevel {
  NONE = 0,
  BASIC = 1,
  INTERMEDIATE = 2,
  ADVANCED = 3,
}

/**
 * Compliance status for an address
 */
export interface ComplianceStatus {
  address: string;
  status: ComplianceStatusType;
  kycLevel: KYCLevel;
  verifiedAt?: Date;
  expiresAt?: Date;
  country?: string;
  riskScore: number;
  sanctioned: boolean;
  flags: string[];
}

/**
 * KYC submission parameters
 */
export interface SubmitKYCParams {
  address: string;
  level: KYCLevel;
  personalInfo: {
    firstName: string;
    lastName: string;
    dateOfBirth: string;
    nationality: string;
  };
  documents: {
    type: string;
    id: string;
    hash: string;
  }[];
  residenceInfo?: {
    country: string;
    city: string;
    postalCode: string;
  };
}

/**
 * Transaction report
 */
export interface TransactionReport {
  txHash: string;
  sender: string;
  recipient: string;
  amount: string;
  denom: string;
  timestamp: Date;
  complianceChecks: {
    sanctionCheck: boolean;
    amlCheck: boolean;
    riskAssessment: number;
  };
  flags: string[];
  approved: boolean;
}

/**
 * Compliance parameters
 */
export interface ComplianceParams {
  kycRequired: boolean;
  sanctionsCheckEnabled: boolean;
  amlEnabled: boolean;
  maxRiskScore: number;
  reportingThreshold: string;
  autoReportEnabled: boolean;
}

/**
 * Sanction check result
 */
export interface SanctionCheckResult {
  address: string;
  sanctioned: boolean;
  lists: string[];
  details?: string;
  checkedAt: Date;
}
